src/librssguard/gui/toolbars/messagestoolbar.cpp

```diff
@@ -125,26 +140,128 @@ void MessagesToolBar::initializeSearchBox() {
+    tr("Filter articles from yesterday"),
+    QVariant::fromValue(MessagesProxyModel::MessageFilter::FilterYesterday),
+    "filter_yesterday");
+  addActionToMenu(m_menuMessageFilter,
+    qApp->icons()->fromTheme(QSL("mail-mark-read")),
+    tr("Filter articles from last 24 hours"),
+    QVariant::fromValue(MessagesProxyModel::MessageFilter::FilterLast24Hours),
+    "filter_last24hours");
+  addActionToMenu(m_menuMessageFilter,
+    qApp->icons()->fromTheme(QSL("mail-mark-read")),
+    tr("Filter articles from last 48 hours"),
+    QVariant::fromValue(MessagesProxyModel::MessageFilter::FilterLast48Hours),
+    "filter_last48hours");
+  addActionToMenu(m_menuMessageFilter,
+    qApp->icons()->fromTheme(QSL("mail-mark-read")),
+    tr("Filter articles from this week"),
+    QVariant::fromValue(MessagesProxyModel::MessageFilter::FilterThisWeek),
+    "filter_this_week");
+  addActionToMenu(m_menuMessageFilter,
+    qApp->icons()->fromTheme(QSL("mail-mark-read")),
+    tr("Filter articles from last week"),
+    QVariant::fromValue(MessagesProxyModel::MessageFilter::FilterLastWeek),
+    "filter_last_week");
+
   m_btnMessageHighlighter = new QToolButton(this);
   m_btnMessageHighlighter->setToolTip(tr("Display all articles"));
   m_btnMessageHighlighter->setMenu(m_menuMessageHighlighter);
   m_btnMessageHighlighter->setPopupMode(QToolButton::ToolButtonPopupMode::MenuButtonPopup);
   m_btnMessageHighlighter->setIcon(qApp->icons()->fromTheme(QSL("mail-mark-read")));
+  m_btnMessageHighlighter->setDefaultAction(m_menuMessageHighlighter->actions().front());
+  m_btnMessageFilter = new QToolButton(this);
+  m_btnMessageFilter->setToolTip(tr("Display all articles"));
+  m_btnMessageFilter->setMenu(m_menuMessageFilter);
+  m_btnMessageFilter->setPopupMode(QToolButton::ToolButtonPopupMode::MenuButtonPopup);
+  m_btnMessageFilter->setIcon(qApp->icons()->fromTheme(QSL("mail-mark-read")));
+  m_btnMessageFilter->setDefaultAction(m_menuMessageFilter->actions().front());
   m_actionMessageHighlighter = new QWidgetAction(this);
   m_actionMessageHighlighter->setDefaultWidget(m_btnMessageHighlighter);
   m_actionMessageHighlighter->setIcon(m_btnMessageHighlighter->icon());
   m_actionMessageHighlighter->setProperty("type", HIGHLIGHTER_ACTION_NAME);
   m_actionMessageHighlighter->setProperty("name", tr("Article highlighter"));
+  m_actionMessageFilter = new QWidgetAction(this);
+  m_actionMessageFilter->setDefaultWidget(m_btnMessageFilter);
+  m_actionMessageFilter->setIcon(m_btnMessageFilter->icon());
+  m_actionMessageFilter->setProperty("type", FILTER_ACTION_NAME);
+  m_actionMessageFilter->setProperty("name", tr("Article filter"));
 
   connect(m_menuMessageHighlighter, &QMenu::triggered, this, &MessagesToolBar::handleMessageHighlighterChange);
+  connect(m_menuMessageFilter, &QMenu::triggered, this, &MessagesToolBar::handleMessageFilterChange);
+}
+
+void MessagesToolBar::saveToolButtonSelection(const QString& buttonName, const QAction* action) const {
```

Comment:
I always use "snake case" notation for local variables, so `button_name` except of `buttonName`, but for class MEMBER variables, I use "pascal notation" with `m_` prefix.

Reply:
Yeah I tried to follow the conventions, I think I missed this one, will follow it from now on